src/agentscope/models/dashscope_model.py

```diff
@@ -330,31 +262,53 @@ def advanced_format(
         """
         # TODO: This function only convert agentscope msgs into qwen
         #  messages, the re-range is executed in _preprocess_role function.
-        prompt = []
-        for unit in args:
-            if isinstance(unit, Msg):
-                prompt.append(f"{unit.name}: {_convert_to_str(unit.content)}")
-            elif isinstance(unit, list):
-                for child_unit in unit:
-                    if isinstance(child_unit, Msg):
-                        prompt.append(
-                            f"{child_unit.name}: "
-                            f"{_convert_to_str(child_unit.content)}",
-                        )
-                    else:
-                        raise TypeError(
-                            f"The input should be a Msg object or a list "
-                            f"of Msg objects, got {type(child_unit)}.",
-                        )
+
+        # Parse all information into a list of messages
+        input_msgs = []
+        for _ in args:
+            if isinstance(_, Msg):
+                input_msgs.append(_)
+            elif isinstance(_, list):
+                input_msgs.extend(_)
             else:
                 raise TypeError(
                     f"The input should be a Msg object or a list "
-                    f"of Msg objects, got {type(unit)}.",
+                    f"of Msg objects, got {type(_)}.",
                 )
 
-        prompt_str = "\n".join(prompt)
+        messages = []
+
+        # record dialog history as a list of strings
+        dialogue = []
+        for i, unit in enumerate(input_msgs):
+            if i == 0 and unit.role == "system":
+                # system prompt
+                messages.append(
+                    {
+                        "role": unit.role,
+                        "content": _convert_to_str(unit.content),
+                    },
+                )
+            else:
+                # Merge all messages into a dialogue history prompt
+                dialogue.append(
+                    f"{unit.name}: {_convert_to_str(unit.content)}",
+                )
 
-        return [{"role": "system", "content": prompt_str}]
+        dialogue_history = "\n".join(dialogue)
+
+        user_content_template = "## Dialogue History\n{dialogue_history}"
+
+        messages.append(
+            {
+                "role": "system",
```

Comment:
If this is the only message for model, the role should be change to "user". Otherwise, it will raise  an error.

Reply:
Done